src/operator/tensor/broadcast_reduce_op.h

```diff
@@ -630,19 +632,32 @@ void SumCsrImpl(const nnvm::NodeAttrs& attrs, mshadow::Stream<xpu>* s, const OpC
                 s, num_threads, output->data().dptr<DType>(), in_indptr, in_idx,
                 in_data, sum.dptr_, residual.dptr_, num_rows, num_cols,
                 seg_len);
+            if (normalize) {
+              mshadow::Tensor<xpu, 1, DType> out_data =
+                  output->data().FlatTo1D<xpu, DType>(s);
+              out_data /= scalar<DType>(num_rows);
+            }
           });
         });
       });
     });
   } else if (1 == param.axis[0]) {
     MSHADOW_IDX_TYPE_SWITCH(input.aux_type(kIndPtr), RType, {
-      MSHADOW_TYPE_SWITCH(input.dtype(), DType, {
-        MXNET_ASSIGN_REQ_SWITCH(req, req_type, {
-          const RType* in_indptr = input.aux_data(kIndPtr).dptr<RType>();
-          const DType* in_data = input.data().dptr<DType>();
-          Kernel<SumCsrKernel<req_type, 1>, xpu>::Launch(
-              s, out_data_size, output->data().dptr<DType>(), in_indptr,
-              in_data);
+      MSHADOW_IDX_TYPE_SWITCH(input.aux_type(kIdx), IType, {
+        MSHADOW_TYPE_SWITCH(input.dtype(), DType, {
+          MXNET_ASSIGN_REQ_SWITCH(req, req_type, {
+            const RType* in_indptr = input.aux_data(kIndPtr).dptr<RType>();
+            const DType* in_data = input.data().dptr<DType>();
+            const IType num_cols = input.shape()[1];
+            Kernel<SumCsrKernel<req_type, 1>, xpu>::Launch(
+                s, out_data_size, output->data().dptr<DType>(), in_indptr,
+                in_data);
+            if (normalize) {
+              mshadow::Tensor<xpu, 1, DType> out_data =
+                  output->data().FlatTo1D<xpu, DType>(s);
+              out_data /= scalar<DType>(num_cols);
```

Comment:
Instead of using mshadow here for the divide, can you use a divide by scalar kernel?  Mshadow is being deprecated.

Reply:
You can use:
Kernel<mxnet_op::op_with_req<mshadow::op__div, Req>, xpu>::Launch(...)

Dee _div_scalar op (elemwise_binary_scalar_op_basic.cc)

Reply:
Thanks Chris! I have fixed this and I am using Kernel::Launch instead of mshadow.